streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -41,37 +53,31 @@ trait ZSinkPlatformSpecificConstructors { self: ZSink.type =>
     path: => Path,
     position: Long = 0L,
     options: Set[OpenOption] = Set(WRITE, TRUNCATE_EXISTING, CREATE)
-  ): ZSink[Blocking, Throwable, Byte, Nothing, Long] =
-    ZSink {
-      for {
-        state <- Ref.make(0L).toManaged_
-        channel <- ZManaged.make(
-                    blocking
-                      .effectBlockingInterrupt(
-                        FileChannel
-                          .open(
-                            path,
-                            options.foldLeft(new ju.HashSet[OpenOption]()) { (acc, op) =>
-                              acc.add(op); acc
-                            } // for avoiding usage of different Java collection converters for different scala versions
-                          )
-                          .position(position)
-                      )
-                      .orDie
-                  )(chan => blocking.effectBlocking(chan.close()).orDie)
-        push = (is: Option[Chunk[Byte]]) =>
-          is match {
-            case None => state.get.flatMap(w => Push.emit(w, Chunk.empty))
-            case Some(byteChunk) =>
-              for {
-                justWritten <- blocking.effectBlockingInterrupt {
-                                channel.write(ByteBuffer.wrap(byteChunk.toArray))
-                              }.mapError(e => (Left(e), Chunk.empty))
-                more <- state.update(_ + justWritten) *> Push.more
-              } yield more
-          }
-      } yield push
+  ): ZSink[Blocking, Throwable, Byte, Nothing, Long] = {
+    val managedChannel = ZManaged.make(
+      blocking
+        .effectBlockingInterrupt(
+          FileChannel
+            .open(
+              path,
+              options.foldLeft(new ju.HashSet[OpenOption]()) { (acc, op) =>
+                acc.add(op); acc
+              } // for avoiding usage of different Java collection converters for different scala versions
+            )
+            .position(position)
+        )
+        .orDie
```

Comment:
I seem to have missed this `orDie` here. Why is it required?

Reply:
It doesn't (I removed it).
Actually, we could reify `Throwable` to `IOException`, but I'm not sure how it would work in case of interruption.